src/d3a/models/area/stats.py

```diff
@@ -50,16 +50,26 @@ def __init__(self, area_markets):
     def update_aggregated_stats(self, area_stats):
         self.aggregated_stats = area_stats
 
+    def _extract_from_bills(self, trade_key):
+        if self.current_market is None:
+            return {}
+        return {key: self.aggregated_stats["bills"][trade_key][key]
+                for key in ["earned", "spent", "bought", "sold"]} \
+            if "bills" in self.aggregated_stats \
+               and trade_key in self.aggregated_stats["bills"] else {}
+
     def update_area_market_stats(self):
         if self.current_market is not None:
             self.market_bills[self.current_market.time_slot] = \
-                {key: self.aggregated_stats["bills"]["Accumulated Trades"][key]
-                 for key in ["earned", "spent", "bought", "sold"]} \
-                if "bills" in self.aggregated_stats \
-                   and "Accumulated Trades" in self.aggregated_stats["bills"] else None
+                {key: self._extract_from_bills(key)
+                 for key in ["Accumulated Trades"]}
             self.rate_stats_market[self.current_market.time_slot] = \
                 self.min_max_avg_median_rate_current_market()
 
+    def get_current_market_stats_for_grid_tree(self):
+        return {key.lower().replace(" ", "_"): self._extract_from_bills(key)
```

Comment:
I assume this is the area name, right? If this is the case I would very much prefer to use slugify for that

Reply:
Now, actually, this is just converting e.g. "Accumulated Trades" into "accumulated_trades" as of a request from Colin.

Reply:
I see, does this apply for area names too though ? Because in this case I could foresee that Colin does not want such a behavior

Reply:
No, this does not apply for area names. If you see one line below, it is only for ["Accumulated Trades", "External Trades"] for now.

Reply:
My sight is getting worse apparently. Thanks a lot for bearing with me!